### PR TITLE
Ci focal rolling

### PR DESCRIPTION
**Pull request by Aposhian**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2813  |
| Primary OS tested on | NA |
| Robotic platform tested on | NA |

---

## Description of contribution in a few bullet points
This is a hotfix to run CI on `ros:rolling-ros-base-focal` while `ros:rolling` is unstable during the transition to Ubuntu 22.04. This circumvents the rosdep keys being updated by avoiding calls to `rosdep update`.

## Description of documentation updates required from your changes
None

---

## Future work that may be required in bullet points
Revert this change when everything is ready to build on Ubuntu 22.04.

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by Aposhian** [comment]
@ruffsl I got this to successfully install dependencies on CircleCI on my fork: https://app.circleci.com/pipelines/github/Aposhian/navigation2/2/workflows/addb9818-071e-406d-89a9-6cd4432d0782/jobs/2

Maybe this is something that can get our CI working while waiting for upstream packages and fixes?

**Comment by ruffsl** [comment]
@Aposhian , clever minimal workaround!

I also thought of just skipping the rosdep update to stick with the cache archived in the old focal image, but figured it'd be better to bite the bullet and at least push through to jammy to keep up with rolling for updates. If we go with this PR, we won't be tracking any further sync updates, as opposed to at least falling back to galactic. But as you mention, such should only be a temporary measure.

As I've recently tried, there are a few API breaks from galactic using tf2 that we've sense adopted, and my attempt to build gazebo_ros from source still hits a snag using jammy: 

 - https://github.com/ros-planning/navigation2/pull/2852#issuecomment-1071957746

I'm fine with this approach, or we could wait to see if the above is easily unblocked. I'll let @SteveMacenski decide though.

**Comment by SteveMacenski** [comment]
@ruffsl I know you don't know the answer to this, but how much longer do you think we'll be waiting for the issues to be resolved upstream of us (minus of course any Gazebo Ignition work we need to do locally to update our stuff to the new simulator)? If we've already sat through the most of it, I'm inclined to just wait it out and communicate our issues to OSRF so they can help unblock.

In the TSC meeting yesterday, Kat mentioned that if we could give her a list of specific problems, she could see what she could do on her side.

**Review by ruffsl — APPROVED** [review]

**Pull request merged**
